wicket-core/src/main/java/org/apache/wicket/validation/validator/DateValidator.java

```diff
@@ -186,7 +188,15 @@ protected IValidationError decorate(IValidationError error, IValidatable<Date> v
 			// format variables if format has been specified
 			if (format != null)
 			{
-				SimpleDateFormat sdf = new SimpleDateFormat(format);
+				WebSession session = WebSession.get();
```

Comment:
First check that there is a Session with `if (Session.exists()) { ... }` because `Session.get()` will create a new one.

Reply:
There is no need of `Web`Session. Just `Session.get()` should be enough to get the locale.